tutorial/chapter1.md

```diff
@@ -1,41 +1,219 @@
+**Application API**. The classes that form the application API are:
 
-* **Composition API**. The classes that form the composition API are:
-    * [`kpsr::vision_ocv::ImageMultiplexerVisionMiddlewareProvider`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/mem_mdlw/include/klepsydra/mem_vision_ocv/image_multiplexer_vision_ocv_provider.h)
-    * [`kpsr::vision_ocv::BasicVisionMiddlewareProvider`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/mem_mdlw/include/klepsydra/mem_vision_ocv/basic_vision_ocv_provider.h)
+- [`kpsr::vision_ocv::ImageDataFactory`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/vision_ocv/include/klepsydra/vision_ocv/image_data_factory.h)
+- [`kpsr::vision_ocv::ImageFileUtils`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/vision_ocv/include/klepsydra/vision_ocv/image_file_utils.h)
+- [`kpsr::vision_ocv::MJPEGWriter`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/vision_ocv/include/klepsydra/vision_ocv/mpeg_writer.h)
+- [`kpsr::vision_ocv::FileImageStreamingService`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/vision_ocv/include/klepsydra/vision_ocv/file_image_stream_service.h)
 
-## Core API example <a name="core-api-example"></a>
+**Composition API**. The classes that form the composition API are:
 
-The [mem example](../examples/mem_example/src/main.cpp) shows the usage of the [`DataMultiplexerMiddlewareProvider`](https://github.com/klepsydra-technologies/kpsr-core/blob/main/high_performance/modules/high_performance/include/klepsydra/high_performance/data_multiplexer_middleware_provider.h) high performance API (from kpsr-core) when processing opencv `Mat` images using `ImageDataFactory` and `ImageMultiplexerVisionMiddlewareProvider` classes. In this sample example, we process images by resizing them to desired size and publish them. After that, the mean of these resized images is calculated and published too.
+- [`kpsr::vision_ocv::ImageMultiplexerVisionMiddlewareProvider`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/mem_mdlw/include/klepsydra/mem_vision_ocv/image_multiplexer_vision_ocv_provider.h)
+- [`kpsr::vision_ocv::BasicVisionMiddlewareProvider`](https://github.com/klepsydra-technologies/kpsr-robotics/blob/main/vision_ocv/modules/mem_mdlw/include/klepsydra/mem_vision_ocv/basic_vision_ocv_provider.h)
+
+## Event loop example 1
+
+This example is the Computer Vision evolution of the [EventLoop example](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/main/tutorials/chapter1.md#example-2-the-eventloop) in the main Klepsydra tutorial. Simulates 3 cameras' images process by resizing and converting to B/W. See the [source code](../examples/event_loop/src/event_loop_example1.cpp) and run:
```

Comment:
Feel free to ignore this comment but the idea was that the number of the cameras was not fixed. The user would hardcode the number of cameras (modifying `totalCameras`), recompile and see that the thread-ids stay the same in example 1 and change for all the odds/even listener in example 2.

Reply:
> This is true for any number of cameras (here `totalCameras=3`).

added in the output paragraph.